File: VB-dotNet/AfricasTalkingGateway/Smspremiumrate.vb
'Sending premium rated messages
Imports System
Imports System.Collections
Friend Class Smspremiumrate
    Public Shared Sub _Smspremiumrate()

        Dim username As String = "MyAfricasTalkingUsername"
        Dim apiKey As String = "MyAfricasTalkingAPIKey"

        Dim recipients As String = "+254711XXXYYY,+254733YYYZZZ"

        Dim message As String = "Get your daily message and that's how we roll."

        ' Specify your premium shortCode and keyword
        Dim shortCode As String = "XXXXX"
        Dim keyword As String = "premiumKeyword"

        ' Set the bulkSMSMode flag to 0 so that the subscriber get charged
        Dim bulkSMSMode As Integer = 0

        ' Create an array which would hold the following parameters:
        ' keyword: Your premium keyword,
        ' retryDurationInHours: The numbers of hours our API should retry to send the message 
        ' incase it doesn't go through. It is optional

        Dim options As New Hashtable()
        options("keyword") = keyword
        options("retryDurationInHours") = "No. of hours to retry sending message"

        Dim gateway As New AfricasTalkingGateway(username, apiKey)

        Try

            Dim results As Object = gateway.sendMessage(recipients, message, shortCode, bulkSMSMode, options)

            For Each result As Object In results
                Console.Write(CStr(result("number")) & ",")
                Console.Write(CStr(result("status")) & ",")
                Console.WriteLine(CStr(result("messageId")))
            Next result
        Catch e As AfricasTalkingGatewayException
            Console.WriteLine("Encountered an error: " & e.Message)
        End Try
    End Sub
End Class

File: ClassLibraryAT/Smspremiumrate.cs
    // Sending premium rated messages
    using System;
    using System.Collections;
    class Smspremiumrate
    {
        static public void _Smspremiumrate(){
            
            string username = "MyAfricasTalkingUsername";
            string apiKey   = "MyAfricasTalkingAPIKey"; 
            
            string recipients = "+254711XXXYYY,+254733YYYZZZ";
            
            string message = "Get your daily message and that's how we roll.";
            
            // Specify your premium shortCode and keyword
            string shortCode = "XXXXX";
            string keyword   = "premiumKeyword";
            
            // Set the bulkSMSMode flag to 0 so that the subscriber get charged
            int bulkSMSMode = 0;
            
            // Create an array which would hold the following parameters:
            // keyword: Your premium keyword,
            // retryDurationInHours: The numbers of hours our API should retry to send the message 
            // incase it doesn't go through. It is optional
            
            Hashtable options = new Hashtable();
            options["keyword"] = keyword;
            options["retryDurationInHours"] = "No. of hours to retry sending message";
            
            AfricasTalkingGateway gateway = new AfricasTalkingGateway (username, apiKey);
            
            try {
                
                dynamic results = gateway.sendMessage (recipients, message, shortCode, bulkSMSMode, options);
                
                foreach( dynamic result  in results){
                    Console.Write((string)result["number"] + ",");
                    Console.Write((string)result["status"] + ",");
                    Console.WriteLine((string)result["messageId"]);
                }
            } catch (AfricasTalkingGatewayException e) {
                Console.WriteLine ("Encountered an error: " + e.Message);        
            }
        }
    }